### PR TITLE
Separate leaves initialization profiling region into two regions

### PR DESCRIPTION
**Pull request by aprokop**
When profiling things, sometimes one has to do arithmetic to figure out
whether it is the Morton codes sorting that slows down, or leaf
initialization. Might as well separate them.

### CODE REVIEW

File: src/ArborX_LinearBVH.hpp [hunk 1]
```diff
@@ -196,17 +196,22 @@ BoundingVolumeHierarchy<DeviceType>::BoundingVolumeHierarchy(
...
       Kokkos::ViewAllocateWithoutInitializing("morton"), size());
   Details::TreeConstruction<DeviceType>::assignMortonCodes(
       primitives, morton_indices, _bounds);
 
   Kokkos::Profiling::popRegion();
-  Kokkos::Profiling::pushRegion("ArborX:BVH:sort_morton_codes_and_init_leaves");
+  Kokkos::Profiling::pushRegion("ArborX:BVH:sort_morton_codes");
 
-  // sort them along the Z-order space-filling curve
+  // sort Morton codes
```
Comment (dalg24) [inline src/ArborX_LinearBVH.hpp]: Undo this change.

**Comment by masterleinad** [comment]
Retest this please.

**Comment by aprokop** [comment]
I tweaked the comments, hopefully they read better now.

File: src/ArborX_LinearBVH.hpp [hunk 2]
```diff
@@ -196,17 +196,22 @@ BoundingVolumeHierarchy<DeviceType>::BoundingVolumeHierarchy(
...
       Kokkos::ViewAllocateWithoutInitializing("morton"), size());
   Details::TreeConstruction<DeviceType>::assignMortonCodes(
       primitives, morton_indices, _bounds);
 
   Kokkos::Profiling::popRegion();
-  Kokkos::Profiling::pushRegion("ArborX:BVH:sort_morton_codes_and_init_leaves");
+  Kokkos::Profiling::pushRegion("ArborX:BVH:sort_morton_codes");
 
-  // sort them along the Z-order space-filling curve
+  // compute the ordering of primitives along Z-order space-filling curve
```
Comment (dalg24) [inline src/ArborX_LinearBVH.hpp]: ... and actually sort the Morton codes ;)

Comment (aprokop) [reply src/ArborX_LinearBVH.hpp]: I give up. Could you pretty please push whatever comments please you?

**Comment by masterleinad** [comment]
The last commit was not tested !? :flushed:

**Comment by aprokop** [comment]
Could be that the network was down again, it was flaky at times yesterday.
